src/core_ocean/shared/mpas_ocn_tracer_advection_mono.F

```diff
@@ -656,46 +659,62 @@ subroutine ocn_tracer_advection_mono_tend(tend, tracers, layerThickness,    &
+          !    being directed upward but increasing k points downward.  This is easily shown by writing the
+          !    3rd order solution (coef3rdOrder = 1), we have
+          !      flux = w(k-1/2)*(0.5*(tracer(k)+tracer(k-1)) - 1/12*(tracer(k+1) - tracer(k) - tracer(k-1) + 
+          !             tracer(k-2)) + sign(w(k-1/2))/12*(tracer(k+1) - 3*tracer(k) + 3*tracer(k-1) - tracer(k-2)))
+          !    There are two cases 
+          !    a. positive w
+          !       flux = w(k-1/2)*(0.5*(tracer(k)+tracer(k-1)) - 1/12*(-2*tracer(k) + 4*tracer(k-1) - 2*tracer(k-2)))
+          !    b. negative w
+          !       flux = w(k-1/2)*(0.5*(tracer(k)+tracer(k-1)) - 1/12*(-2*tracer(k+1) + 4*tracer(k) - 2*tracer(k-1)))
+          !    in the ocean, positive w is upward, but would be pulling layers downstream, hence we need the negative 
+          !    from the original SG11 equation (9)
+          ! 2. abs(w) appears where sign(w) appears in SG11.  Recall w = abs(w)*sign(w).  in this 
+          !    implementation w is multiplied through so we would have w*sign(w), using the formula we have
+          !    abs(w)*sign(w) / sign(w) = abs(w)
+          ! Relevant reference - Skamarock, W. C., & Gassmann, A. (2011). Conservative transport schemes for 
+          ! spherical geodesic grids: High-order flux operators for ODE-based time integration. 
+          ! Monthly Weather Review, 139(9), 2962-2975.
           !$omp parallel
-          !$omp do schedule(runtime) private(kmax, k)
+          !$omp do schedule(runtime) private(kmax, k, dzEdge, deriv2K, deriv2Km1)
           do iCell = 1, nCells
             kmax = maxLevelCell(iCell)
             do k=3,kmax-1
-              highOrderFlx(k, iCell) = (w(k,iCell)* &
-                   (7.0_RKIND*(tracerCur(k  ,iCell) + tracerCur(k-1,iCell)) - &
-                              (tracerCur(k+1,iCell) + tracerCur(k-2,iCell))) - &
-                        coef3rdOrder*abs(w(k,iCell))* &
-                             ((tracerCur(k+1,iCell) - tracerCur(k-2,iCell)) - &
-                    3.0_RKIND*(tracerCur(k  ,iCell) - tracerCur(k-1,iCell))))/ &
-                   12.0_RKIND
-
-
+              deriv2K = (tracerCur(k-1,iCell) - tracerCur(k,iCell))/(0.5_RKIND*hProv(k,iCell)* &
+                        (hProv(k-1,iCell) + hProv(k,iCell))) - (tracerCur(k,iCell) - tracerCur(k+1,iCell))/ &
+                        (0.5_RKIND*hProv(k,iCell)*(hProv(k,iCell) + hProv(k+1,iCell)))
+              deriv2Km1 = (tracerCur(k-2,iCell) - tracerCur(k-1,iCell))/(0.5_RKIND*hProv(k-1,iCell)* &
+                        (hProv(k-2,iCell) + hProv(k-1,iCell))) - (tracerCur(k-1,iCell) - tracerCur(k,iCell)) / &
+                        (0.5_RKIND*hProv(k-1,iCell)*(hProv(k-1,iCell) + hProv(k,iCell)))
+              dzEdge = 0.5*(hProv(k-1,iCell) + hProv(k,iCell))
+
+              highOrderFlx(k, iCell) = w(k,iCell)*(0.5_RKIND*(tracerCur(k,iCell) + tracerCur(k-1,iCell)) - &
```

Comment:
I think the primary spot where using hProv weighting will benefit the flux is in the 2nd-order part. Here, you've kept this as `1/2 (q(k) + q(k-1))` and used hProv only in the second-derivative estimates. The vertOrder2 code uses the hProv-weighted version based on a linear interpolant; I think you should bring this into this expression.

That aside, I think we should be cautious about making changes to this code without a vertical-direction convergence test.

Reply:
Thanks @ambrad you're right, I didn't even think about that.  It is of course obvious that part is fixed grid too.

I also agree we should have a vertical advection convergence test.  I started implementing that test with a postbac but didn't get it fully cleaned up.